proxytools/scrappers/premproxy.py

```diff
@@ -0,0 +1,164 @@
+                                ips.append(url)
+
+        for ip in ips:
+            ip = ip.strip()
+            if ip and validate_ip(ip.split(':')[0]):
+                proxylist.append('http://{}'.format(ip))
+
+        log.info('Parsed %d socks5 proxies from webpage.', len(proxylist))
+        return proxylist
+
+    # Premproxy does not have a consistent number of additional pages.
+    # Check the main page for the links for pages and extract from there.
+    def extract_pages(self):
+
+        urls = []
+        page_urls = []
+        list_url = self.base_url + '/list/'
+
+        html = self.request_url(list_url)
+        if html is None:
+            log.error('Failed to download webpage: %s', list_url)
+            return urls
+
+        # Check to see how many additional pages we can grab.
+        soup = BeautifulSoup(html, 'html.parser')
+        # soup.prettify()
+
+        pagination = soup.find('ul', class_='pagination')
+        if not pagination:
+            log.error('Unable to find element with proxy list.')
+            return urls
+
+        links = pagination.findAll("a")
+        for link in links:
+            if not link.get_text() == 'next':
+                urls.append(link.get('href'))
+
+        # Now go through each of the page links and create the final url.
+        for url in urls:
+            if "list" in url:
+                page_urls.append(self.base_url + url)
+            else:
+                page_urls.append(list_url + url)
+
+        return page_urls
+
+    # Premproxy.com uses javascript to obfuscate proxies port number.
+    # Builds a dictionary with decoded values for each variable.
+    # Dictionary = {'var': intValue, ...})
+    def extract_ports(self, js_url):
+
+        dictionary = {}
+
+        # Download the JS file.
+        html = self.request_url(js_url)
+        if html is None:
+            log.error('Failed to download webpage: %s', js_url)
+            return dict
+
+        try:
+            # For now, try and extract out the css/port pairs from the JS.
+            # This likely is a really back way of doing it, I'll revisit later.
+            unpack = packer.unpack(html)
+            unpack = unpack.replace("$(document).ready(function(){", "")
+            unpack = unpack.replace("});", "")
+            unpack = unpack.replace("\\", "")
+            unpack = unpack.replace("'", "")
+            unpack = unpack.replace(".", "")
+
+            # Pull out everything that is within a bracket.
+            parts = re.findall('\((.*?)\)', unpack)
+
+            # Now convert the list into a dictionary.
+            # Every other entry in the list is a pair css/port.
+            i = 0
+            while i < len(parts):
+                dictionary[parts[i]] = parts[i+1]
+                i += 2
+
+            return dictionary
+
+        except Exception as e:
+            log.exception('Failed do extract ports from %s: %s.', js_url, e)
+            return dictionary

```

Comment:
You can decrement indentation of this last line (`return dictionary`) and remove the `return dictionary` from line 160.